examples/resources/kentik-synthetics_test/resource.tf

```diff
@@ -0,0 +1,85 @@
+resource "kentik-synthetics_test" "example-hostname-test" {
+  name      = "example-hostname-test"
+  type      = "hostname"
+  device_id = "75702"
+  status    = "TEST_STATUS_ACTIVE"
+  settings {
+    hostname {
+      target = "www.example.com"
+    }
+    agent_ids = tolist([
+      "817",
+      "818",
+      "819"
+    ])
+    period = 61
+    count  = 2
+    expiry = 5001
+    limit  = 4
+    tasks = tolist([
+      "ping",
+      "traceroute"
+    ])
+    health_settings {
+      latency_critical      = 1
+      latency_warning       = 2
+      packet_loss_critical  = 3
+      packet_loss_warning   = 4
+      jitter_critical       = 5
+      jitter_warning        = 6
+      http_latency_critical = 7
+      http_latency_warning  = 8
+      http_valid_codes = tolist([
+        200,
+        201
+      ])
+      dns_valid_codes = tolist([
+        1,
+        2,
+        3
+      ])
+    }
+    monitoring_settings {
+      activation_grace_period = "2"
+      activation_time_unit    = "m"
+      activation_time_window  = "5"
+      activation_times        = "3"
+      // Notice: currently "notification_channels" field cannot be manipulated
+      //      notification_channels = tolist([
+      //        "dummy-channel-1",
+      //        "dummy-channel-2",
+      //      ])
+    }
+    ping {
+      period = 60
+      count  = 5
+      expiry = 3000
+    }
+    trace {
+      period   = 60
+      count    = 3
+      protocol = "udp"
+      port     = 33434
+      expiry   = 22500
+      limit    = 30
+    }
+    port     = 443
+    protocol = "icmp"
+    family   = "IP_FAMILY_V6"
+    servers = tolist([
+      "server-one",
+      "server-two",
+    ])
+    // Notice: specific values expected
+    // TODO(dfurman): provide allowed values
+    // target_type  = ""
```

Comment:
If I remember correctly, these two were read-only - server would reject the request carrying these params set

Reply:
Server responds as following:

> error: 400 Bad Request, status: 400 Bad Request, body: {{"code":3,"message":"At path: request.test.settings.target_type -- Expected a value of type `never`, but received: `\"foo\"`","details":[]}}


It look like client needs to pass specific value. @mmac-m3a Could you find out what are allowed values?

Reply:
I think Mateusz is right. The error message confirms it. It says it expect value of "type never", which is it's way to say that it is not expecting the attribute to be preset. I'll verify with Kentik engineering. Daniel, can you give me the complete request?

Reply:
This is the request that triggers the 400 response:
```bash
curl --location --request POST 'https://synthetics.api.kentik.com/synthetics/v202101beta1/tests' \
--header 'Content-Type: application/json' \
--header 'X-CH-Auth-Email: <email>' \
--header 'X-CH-Auth-API-Token: <token>' \
--data-raw '{
    "test": {
        "deviceId": "75702",
        "name": "example-hostname-test",
        "settings": {
            "agentIds": [
                "817",
                "818",
                "819"
            ],
            "count": 2,
            "expiry": 5001,
            "family": "IP_FAMILY_V6",
            "healthSettings": {
                "dnsValidCodes": [
                    1,
                    2,
                    3
                ],
                "httpLatencyCritical": 7,
                "httpLatencyWarning": 8,
                "httpValidCodes": [
                    200,
                    201
                ],
                "jitterCritical": 5,
                "jitterWarning": 6,
                "latencyCritical": 1,
                "latencyWarning": 2,
                "packetLossCritical": 3,
                "packetLossWarning": 4
            },
            "hostname": {
                "target": "www.example.com"
            },
            "limit": 4,
            "monitoringSettings": {
                "activationGracePeriod": "2",
                "activationTimeUnit": "m",
                "activationTimeWindow": "5",
                "activationTimes": "3",
                "notificationChannels": []
            },
            "period": 61,
            "ping": {
                "count": 5,
                "expiry": 3000,
                "period": 60
            },
            "port": 443,
            "protocol": "icmp",
            "reciprocal": false,
            "rollupLevel": 1,
            "servers": [
                "server-one",
                "server-two"
            ],
            "targetType": "foo",
            "targetValue": "bar",
            "tasks": [
                "ping",
                "traceroute"
            ],
            "trace": {
                "count": 3,
                "expiry": 22500,
                "limit": 30,
                "period": 60,
                "port": 33434,
                "protocol": "udp"
            },
            "useLocalIp": true
        },
        "status": "TEST_STATUS_ACTIVE",
        "type": "hostname"
    }
}'
```